packages/wonder-blocks-announcer/src/index.ts

```diff
@@ -0,0 +1,334 @@
+// TODO: publish wonder-blocks-style package WB-1776
+// import {srOnly} from "../../wonder-blocks-style/src/styles/a11y";
+import type {
+    PolitenessLevel,
+    RegionDef,
+    RegionDictionary,
+    RegionFactory,
+} from "./announcer.types";
+
+const TIMEOUT_DELAY = 5000;
+let announcer: Announcer | null = null;
+
+export type SendMessageProps = {
+    message: string;
+    level?: PolitenessLevel;
+    removalDelay?: number;
+};
+
+/**
+ * Public API method to send screen reader messages.
+ * @param {string} message The message to send.
+ * @param {PolitenessLevel} level Polite or assertive announcements
+ * @param {number} removalDelay Optional duration to remove a message after sending. Defaults to 5000ms.
+ * @returns {string} IDREF for targeted live region element
+ */
+export function sendMessage({
+    message,
+    level = "polite", // TODO: decide whether to allow other roles, i.e. role=`timer`
+    removalDelay,
+}: SendMessageProps): string {
+    announcer = Announcer.getInstance();
+
+    if (typeof jest === "undefined") {
+        setTimeout(() => {
+            return announcer?.announce(message, level, removalDelay);
+        }, 100);
+    } else {
+        // If we are in a test environment, announce without waiting
+        return announcer.announce(message, level, removalDelay);
+    }
+    return "";
+}
+/**
+ * Public API method to clear screen reader messages after sending.
+ * Clears all regions by default.
+ * @param {string} id Optional id of live region element to clear.
+ */
+export function clearMessages(id?: string) {
+    if (id && document?.getElementById(id)) {
+        announcer?.clear(id);
+    } else if (document) {
+        announcer?.clear();
+    }
+}
+
+/**
+ * Internal class to manage screen reader announcements.
+ */
+class Announcer {
+    private static _instance: Announcer;
+    node: HTMLElement | null = null;
+    regionFactory: RegionFactory = {
+        count: 2,
+        aIndex: 0,
+        pIndex: 0,
+    };
+    dictionary: RegionDictionary = new Map();
+
+    private constructor() {
+        if (typeof document !== "undefined") {
+            const topLevelId = `wbAnnounce`;
+            // Check if our top level element already exists
+            const announcerCheck = document.getElementById(topLevelId);
+
+            // Init new structure if the coast is clear
+            if (announcerCheck === null) {
+                this.init(topLevelId);
+            }
+            // The structure exists but references are lost, so help HMR recover
+            else {
+                this.rebootForHMR();
```

Comment:
**thought:** ohh I see, yeah this is annoying. I wonder if you could setup storybook to initiate Announcer outside of the stories themselves. Not sure if this would be useful:
https://storybook.js.org/docs/api/main-config/main-config-preview-body

I haven't tried it out yet, but I was thinking about creating an initial DOM element to attach all the announcer contents to it. Similar to how react uses `createRoot` (e.g. `<div id="root">`). I'm not sure if you have explored a similar solution to this or not, but it could be worth considering.